### PR TITLE
Feature database redesign

### PR DESCRIPTION
**Pull request by rohanbatra24**
## What additional changes are needed to implement this feature (please keep this as a high level summary)

- To implement the group chat feature, the frontend logic needs to be updated to use Socket.io's built-in  'Rooms' functionality. 
- In our case, a room will be a group chat. With this functionality, members will have the ability to join and leave chats
- Since we have updated the backend database model (added a 'Participant' table to join Users and Conversations), we can now keep track of all group chat members and show in UI accordingly
- The UI needs to be updated to add ability to create new group chats/add members to existing chat

## If our app were already deployed, what steps would we need to take to move to this new feature without disrupting service for current users

- Before deploying this change to production, we would need to run database queries on our existing models to ensure that the current data (members, messages, conversations) is compliant with the new models and their associations
- This would require modifying the table rows and adding data to those columns which now have additional constraints (not null, foreign keys etc.) based on the new models/assocations
- For example, since our conversation table now has a mandatory 'Name' column, we would need to run UPDATE queries to add a default name to each existing convo so that our app doesn't break when deployed
- To avoid disruption of service, we could use a load balancer and run two identical instances of the app. When deploying the new feature, we could tell the load balancer to only forward the traffic to one instance while we update the other one and then switch the traffic to the other once deployed. This way, the app will remain live for users
- Alternatively, we could take the app down while deploying and show a 'maintenance' screen during that time

## New ERD
Screen Shot 2021-08-23 at 5 53 50 PM

### CODE REVIEW

**Review by sonipranjal — COMMENTED** [review]
Great work! Please address the comments below and let us know when this is ready for a second review!

**Review by FilipeGorodscy — APPROVED** [review]
Looks good to me, I just have one question:
- What are the changes (if any) that you would need to implement to handle this new feature?

Approved, but don't merge! :)

**Pull request closed without merging**
